### DISCUSSION TITLE
Is there a way to hide a column in <DataTable> depending on mobile/desktop?

### DISCUSSION THREAD

**Discussion by AnthonyVuOD** [primefaces/community#1391] [PrimeReact]
Is there a way to hide a column in <DataTable> when on mobile and show all columns on desktop? I am trying to make my application more responsive.

**Comment by BookerE1** [comment]
Certainly! To make your <DataTable> responsive and hide specific columns based on the device screen size (mobile or desktop), you can follow these steps:

Include Required Scripts:
First, ensure that you include the necessary scripts for responsive behavior. Add the following to your HTML:
HTML


<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/responsive/2.2.3/css/responsive.dataTables.min.css">
<script type="text/javascript" src="https://cdn.datatables.net/responsive/2.2.3/js/dataTables.responsive.min.js"></script>

<script type="text/javascript" src="https://cdn.datatables.net/colreorder/1.5.2/js/dataTables.colReorder.min.js"></script>

Add Display Classes to Columns:
To hide specific columns, add a display class to those columns. You can use the min-phone-l class to hide columns on small screens (mobile devices). For example:
HTML

<table id="myDataTable">
    <thead>
        <tr>
            <th></th> 
            <th></th> 
            <th class="min-phone-l"></th> 
            <th class="min-phone-l"></th> 
            <th data-priority="1"></th> 
        </tr>
    </thead>
    
</table>

Column Reordering (Optional): [DogNeedsBest](https://www.dogneedsbest.com/)
If you want to allow users to reorder columns, consider using the DataTables column reordering plugin. You can find more details about it here.

Test and Verify:
Test your responsive DataTable on both mobile and desktop screens to ensure that the columns behave as expected.
Remember to adjust the column classes and priorities according to your specific use case. With these steps, you’ll be able to create a responsive DataTable that adapts to different screen sizes!
